src/ecmascript_simd.js

```diff
@@ -474,13 +474,13 @@ if (typeof SIMD.Bool32x4.select === "undefined") {
   }
 }
 
-if (!Object.hasOwnProperty(SIMD.Bool32x4.prototype, 'valueOf')) {
+if (!SIMD.Bool32x4.prototype.hasOwnProperty('valueOf')) {
```

Comment:
That's a fair point. Since this is instantly shimming the global (instead of exporting a function to do that on demand), they're basically the same.

If you had the not-auto-shim behavior, you'd do:
```js
var has = Function.call.bind(Object.prototype.hasOwnProperty);
module.exports = function shim() {
 …
 if (has(SIMD.Bool32x4.prototype, 'valueOf')) {
    …
 }
 …
};
```